pkg/apis/autoscaling/v1/fleetautoscaler.go

```diff
@@ -429,7 +429,7 @@ func (c *CounterPolicy) ValidateCounterPolicy(causes []metav1.StatusCause) []met
 			causes = append(causes, metav1.StatusCause{
 				Type:    metav1.CauseTypeFieldValueInvalid,
 				Field:   "bufferSize",
-				Message: "bufferSize must be bigger than 0",
+				Message: "bufferSize must be bigger than 0", // TODO: Should we allow bufferSize to be Zero (in case of Scaling down to 0)? In case of BufferPercentage MaxCapacity can be set to 0, in case of BufferInt Max Capacity must be >= 2.
```

Comment:
Should we allow bufferSize to be Zero (in case of Scaling down to 0)? In case of BufferPercentage MaxCapacity can be set to 0, in case of BufferInt Max Capacity must be >= 2.

Reply:
It's not something we can support in autoscaling. If it's zero and they make an allocation request, there is nowhere for it to go, since everything is zero.

This is why we have webhooks, because then the end user can use other signals - login services, matchmakers etc to let them know they need to scale up GameServers. As Agones, we don't get any of that information.

Reply:
Does your comment "If it's zero and they make an allocation request, there is nowhere for it to go" refer to the buffer? Or the desired number of replicas? Just confirming because we can reach a desired number of replicas == 0.

Reply:
Per discussion we do not allow autoscaling to Zero (unless by custom Webhook).